### PR TITLE
feat(ConditionBuilder): enhancing the conditional operators section that manages the primary logic flow

### LINKED ISSUE #5858: ConditionBuilder: Enhancing the conditional operators section that manages the primary logic flow
Changing the section as below

**old** 
image

 **new**
image

### PR DESCRIPTION
**Pull request by amal-k-joy**
Closes #5858

Enhancing the conditional operators section that manages the primary logic flow
Moved the data configs to a new hook useDataConfig to leverage useTranslations hook for operators
Moved all the util functions and constants to util.js

#### How did you test and verify your work?
local storybook

### CODE REVIEW

File: packages/ibm-products/src/components/ConditionBuilder/utils/util.js [hunk 1]
```diff
@@ -102,3 +106,43 @@ export const manageTabIndexAndFocus = (currentElement, conditionBuilderRef) => {
...
 };
+
+const formatDate = (date) => {
+  const day = String(date.getDate()).padStart(2, '0');
+  const month = String(date.getMonth() + 1).padStart(2, '0');
+  const year = date.getFullYear();
+  return `${day}/${month}/${year}`;
+};
+
+export const valueRenderers = {
```
Comment (davidmenendez) [inline packages/ibm-products/src/components/ConditionBuilder/utils/util.js:117]: 2 minor things. i think we can simplify this to just be something like `getValue` and also for consistency lets stick between either `val` or `value`

**Review by davidmenendez — CHANGES_REQUESTED** [review]
looks good. one minor thing

File: packages/ibm-products/src/components/ConditionBuilder/utils/util.js (hunk 1 above)
Comment (amal-k-joy) [reply packages/ibm-products/src/components/ConditionBuilder/utils/util.js]: modified as advised

File: packages/ibm-products/src/components/ConditionBuilder/ConditionBuilderButton/ConditionBuilderButton.tsx [hunk 2]
```diff
@@ -84,7 +84,7 @@ export const ConditionBuilderButton = ({
   return hideLabel || showToolTip ? (
     <Tooltip
       label={label}
-      align={tooltipAlign}
+      align={tooltipAlign as PopoverAlignment}
```
Comment (AlexanderMelox) [inline packages/ibm-products/src/components/ConditionBuilder/ConditionBuilderButton/ConditionBuilderButton.tsx:87]: Hey @amal-k-joy, is there a reason we don't type it as `PopoverAlignment` in the `ConditionBuilderButtonProps`?

Screenshot 2024-08-23 at 11 07 33 AM

**Review by AlexanderMelox — APPROVED** [review]

File: packages/ibm-products/src/components/ConditionBuilder/ConditionBuilderButton/ConditionBuilderButton.tsx (hunk 2 above)
Comment (amal-k-joy) [reply packages/ibm-products/src/components/ConditionBuilder/ConditionBuilderButton/ConditionBuilderButton.tsx]: Thanks @AlexanderMelox , Good catch

**Review by davidmenendez — APPROVED** [review]

**Review by AlexanderMelox — APPROVED** [review]

**Pull request merged**